components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java

```diff
@@ -237,11 +239,19 @@ private String replaceFromCodeId(String authzCode) {
      * @return TOKEN_ID from the database
      */
     private String replaceFromTokenId(String keyValue) {
-        try {
-            AccessTokenDO accessTokenDO = OAuth2Util.findAccessToken(keyValue, true);
-            if (accessTokenDO != null) {
-                return accessTokenDO.getTokenId();
+        if (OAuth2Util.isJWT(keyValue)) {
+            try {
+                JWT parsedJwtToken = JWTParser.parse(keyValue);
+                keyValue = parsedJwtToken.getJWTClaimsSet().getJWTID();
+            } catch (ParseException e) {
+                if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable(
+                        IdentityConstants.IdentityTokens.ACCESS_TOKEN)) {
+                    log.debug("Error while getting JWTID from token: " + keyValue, e);
```

Comment:
Shall we add another logs when token is not loggable, otherwise we can't see any logs when the token loggable is disabled? And shouldn't this be a warn log instead of debug log?

Reply:
Resolved with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2613/commits/70a2459b45973367d6f4e2db20d08dfadd25a394